app/services/measurements_creator.rb

```diff
@@ -22,6 +22,7 @@ def call(stream, measurements_attributes)
     return if stream.session.type == 'FixedSession'
     streams_repository.calc_bounding_box!(stream)
     streams_repository.calc_average_value!(stream)
+    streams_repository.add_start_coordinates!(stream)
```

Comment:
This is prolly not changing much but just so we are aware: this adds another roundtrip to database. Maybe it would be better to do all calculations first (bounding box, average, coordinates) and save at once That is whenever possible